modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -262,7 +264,24 @@ public AndesMessageMetadata shallowCopy(long messageId) {
     public void updateMetadata(String newDestination, String newExchangeName) {
         this.metadata = createNewMetadata(this.metadata, newDestination, newExchangeName);
         this.destination = newDestination;
-        log.debug("updated andes message metadata id= " + messageID + " new destination = " + newDestination);
+        if (log.isDebugEnabled()) {
+            log.debug("updated andes message metadata id= " + messageID + " new destination = " + newDestination);
+        }
+    }
+
+    /**
+     * Update metadata of message, after a change of the compression state of the message. This will change AMQP bytes
+     * representing metadata. IsCompressed will be set to the given value.
+     *
+     * @param newCompressedMessageValue new value to indicate if the message is compressed or not
+     */
+    public void updateMetadata(boolean newCompressedMessageValue) {
```

Comment:
'newCompressedMessageValue' doesn't sound like a boolean.. Any other name which could be used i.e something like 'compressionEnabled'